common/bufferedpipe/bufferedpipe.go

```diff
@@ -0,0 +1,49 @@
+package bufferedpipe
+
+import (
+	"fmt"
+	"io"
+)
+
+type pipe struct {
+	bufs chan []byte
+	io.Reader
+	closer io.Closer
+}
+
+// New destroys pipes.  No wait, that can't be right.
+func New() io.ReadWriteCloser {
+	r, w := io.Pipe()
+	p := &pipe{
+		bufs:   make(chan []byte, 100),
+		Reader: r,
+		closer: r,
+	}
+	go p.loop(w)
+	return p
+}
+
+func (p *pipe) Write(b []byte) (int, error) {
+	// copy the buffer
+	buf := make([]byte, len(b))
+	copy(buf, b)
+
+	select {
+	case p.bufs <- buf:
+		return len(buf), nil
+	default:
+		return 0, fmt.Errorf("Pipe full")
```

Comment:
Is there a standard error we can use here? Or variable-ize this for comparisons...